### DISCUSSION TITLE
Keda pods not working with cluster autoscaler

### DISCUSSION THREAD

**Discussion by subhamkhemka** [kedacore/keda#1985] [Q&A / Need Help] [answered]
Hi

I have installed Keda version 2.2.0

Job Yaml

`apiVersion: keda.sh/v1alpha1
kind: ScaledJob
metadata:
  name: scaler-1
  namespace: default
  labels:
    test: scaler-1
spec:
  pollingInterval: 30  # Optional. Default: 30 seconds
  maxReplicaCount: 100 # Optional. Default: 100
  scalingStrategy:
    strategy: accurate                        # Optional. Default: default. Which Scaling Strategy to use. 
  jobTargetRef:
    parallelism: 1
    completions: 1
    activeDeadlineSeconds: 12000
    backoffLimit: 6
    template:
      spec:
        nodeSelector:
          node-type: dev-1
        containers:
        - name: segmentation-model
          image: model:v1
          resources:
            limits:
              memory: "16Gi"
              cpu: "2000m"
            requests:
              memory: "12Gi"
              cpu: "1500m"
        terminationGracePeriodSeconds: 30
  triggers:
  - type: aws-sqs-queue
    authenticationRef:
      name: keda-c360-trigger
    metadata:
      queueURL: "url"
      awsRegion: "us-east-1"
      awsRoleArn: AWS_ROLE_ARN
      queueLength: "1"
`
We get error -

error: error validating "segmentation-new.yaml": error validating data: ValidationError(ScaledJob.spec): unknown field "scalingStrategy" in sh.keda.v1alpha1.ScaledJob.spec; if you choose to ignore these errors, turn validation off with --validate=false

When we try with --validate=False it works

It also works without any error when we remove
`scalingStrategy:
    strategy: accurate                        # Optional. Default: default. Which Scaling Strategy to use. `

The scaled jobs are created and the pods appear when messages are in queue.

However they are not scheduled using cluster autscaler

Logs from cluster autoscaler

I0728 12:20:59.506878       1 klogx.go:86] Pod default/c360-segmentation-r5large-jwc52-zzrcj is unschedulable
I0728 12:20:59.506882       1 klogx.go:86] Pod qa-c360/c360-models-858675d5b8-rhlzf is unschedulable
I0728 12:20:59.506920       1 scale_up.go:364] Upcoming 0 nodes
I0728 12:20:59.506962       1 scale_up.go:288] Pod c360-models-858675d5b8-rhlzf can't be scheduled on eks-2cbd5728-e73b-e521-ec18-55c10e58f719, predicate checking error: node(s) didn't match Pod's node affinity; predicateName=NodeAffinity; reasons: node(s) didn't match Pod's node affinity; debugInfo=
I0728 12:20:59.506979       1 scale_up.go:288] Pod eks-c360-node-failure-handler-75544c6967-plncq can't be scheduled on eks-2cbd5728-e73b-e521-ec18-55c10e58f719, predicate checking error: node(s) didn't match Pod's node affinity; predicateName=NodeAffinity; reasons: node(s) didn't match Pod's node affinity; debugInfo=
I0728 12:20:59.506999       1 scale_up.go:288] Pod c360-segmentation-r5large-jwc52-zzrcj can't be scheduled on eks-2cbd5728-e73b-e521-ec18-55c10e58f719, predicate checking error: Insufficient cpu; predicateName=NodeResourcesFit; reasons: Insufficient cpu; debugInfo=
I0728 12:20:59.507028       1 scale_up.go:437] No pod can fit to eks-2cbd5728-e73b-e521-ec18-55c10e58f719
I0728 12:20:59.507061       1 scale_up.go:288] Pod c360-models-858675d5b8-rhlzf can't be scheduled on eks-b6bd5728-5461-28b1-924a-b3a04318fe8f, predicate checking error: node(s) didn't match Pod's node affinity; predicateName=NodeAffinity; reasons: node(s) didn't match Pod's node affinity; debugInfo=
I0728 12:20:59.507074       1 scale_up.go:288] Pod eks-c360-node-failure-handler-75544c6967-plncq can't be scheduled on eks-b6bd5728-5461-28b1-924a-b3a04318fe8f, predicate checking error: node(s) didn't match Pod's node affinity; predicateName=NodeAffinity; reasons: node(s) didn't match Pod's node affinity; debugInfo=
I0728 12:20:59.507090       1 scale_up.go:288] Pod c360-segmentation-r5large-jwc52-zzrcj can't be scheduled on eks-b6bd5728-5461-28b1-924a-b3a04318fe8f, predicate checking error: node(s) didn't match Pod's node affinity; predicateName=NodeAffinity; reasons: node(s) didn't match Pod's node affinity; debugInfo=
I0728 12:20:59.507117       1 scale_up.go:437] No pod can fit to eks-b6bd5728-5461-28b1-924a-b3a04318fe8f
I0728 12:20:59.507142       1 scale_up.go:400] Skipping node group eks-f2bd5729-cc50-c85b-a87a-803d53fc21c1 - max size reached

We have added following tags to our aws ASG

k8s.io/cluster-autoscaler/cluster-name | owned | Yes
k8s.io/cluster-autoscaler/enabled | true | Yes
k8s.io/cluster-autoscaler/node-template/label/node-type | dev-1 | Yes
kubernetes.io/cluster/cluster-name | owned | Yes

Kubernetes Labels added

k8s.io/cluster-autoscaler/node-template/label/node-type | dev-1 
Kindly help

Thanks,
Subham

**Comment by zroubalik** [comment, marked as answer]
Hi seems like you have an old version of ScaledJob CRD installed in the cluster, so it doesn't recognize the newly introduced `scalingStrategy` option.
I recommend you to do full unistall of KEDA and check that there aren't any leftover CRDs (`kubectl get crd | grep keda`) and then do the full installation again.

The latest version is KEDA 2.3 - I recommend you to use this one and btw we will release 2.4 in a couple of weeks.

**Reply by subhamkhemka** [reply]
Uninstalling keda completely  and installing version 2.3 fixed the issue.

Thanks @zroubalik
